samples/wfi32iot/basic-sample/src/iotconnect_app.c

```diff
@@ -423,17 +609,30 @@ bool iotconnect_sample_app(NX_IP *ip_ptr, NX_PACKET_POOL *pool_ptr, NX_DNS *dns_
             printf("Unable to establish the IoTConnect connection.\r\n");
             return false;
         }
-        // send telemetry at regular intervals
-        for (int i = 0; i < 36000; i++) { // 10 hours if 1s delay
-            if (iotconnect_sdk_is_connected()) {
-                publish_telemetry();  // underlying code will report an error
-                iotconnect_sdk_poll(1000);
-            } else {
+        //MAIN LOOP
+        while (1) 
+        {
+            if (iotconnect_sdk_is_connected()) 
+            {
+                publish_telemetry(warming_up, connections);
+                //Yellow LED Blinks for 500ms when telemetry is being sent
+                LED_YELLOW_On();
+                iotconnect_sdk_poll(500);
+                LED_YELLOW_Off();
+                iotconnect_sdk_poll(loop_delay);
+                //Maximum warm-up time is 15 minutes
+                if(warming_up < 181)
```

Comment:
Maybe better give a rough milit like 1000 and comment about rolling over. You can also make this a unit and it will need 60 years to roll over, so you really don't even need to worrry about it.